lib/java/com/google/android/material/bottomsheet/BottomSheetBehavior.java

```diff
@@ -796,10 +800,14 @@ public void onStopNestedScroll(
             }
           }
         } else {
-          if (Math.abs(currentTop - halfExpandedOffset) < Math.abs(currentTop - collapsedOffset)) {
-            targetState = STATE_HALF_EXPANDED;
-          } else {
+          if (shouldSkipHalfExpandedStateWhenDragging()) {
```

Comment:
if (shouldSkipHalfExpandedStateWhenDragging() || Math.abs(currentTop - halfExpandedOffset) >= Math.abs(currentTop - collapsedOffset)) {

to avoid the below nested if

Reply:
Completed in 7564954